ui/src/treeDataProvider/AzExtTreeItem.ts

```diff
@@ -74,7 +74,7 @@ export abstract class AzExtTreeItem implements types.AzExtTreeItem {
     //#region Methods implemented by base class
     public refreshImpl?(): Promise<void>;
     public isAncestorOfImpl?(contextValue: string | RegExp): boolean;
-    public deleteTreeItemImpl?(deleteTreeItemImpl: types.IActionContext): Promise<void>;
+    public deleteTreeItemImpl?(deleteTreeItemImpl: types.IActionContext): Promise<boolean>;
```

Comment:
As it is, I believe this would force everyone to update `deleteTreeItemImpl`. I believe we can make your new behavior optional. I don't use method overloading all that often in typescript, but something like this might work:
```
public deleteTreeItemImpl?(context: IActionContext): Promise<boolean>;
public deleteTreeItemImpl?(context: IActionContext): Promise<void>;
```

Not sure if the type of `result` at that point should be `boolean | void` or `boolean | undefined` but either way I think it'd work.
```
const result: boolean | void = await this.deleteTreeItemImpl(context);
```

Reply:
(See [reply](https://github.com/microsoft/vscode-azuretools/pull/671#discussion_r394540733) to Steven below)